src/stores/room-list/RoomListStore2.ts

```diff
@@ -64,6 +60,7 @@ export class RoomListStore2 extends AsyncStore<ActionPayload> {
         this.checkEnabled();
         for (const settingName of this.watchedSettings) SettingsStore.monitorSetting(settingName, null);
         RoomViewStore.addListener(this.onRVSUpdate);
+        this.algorithm.on(LIST_UPDATED_EVENT, this.onAlgorithmListUpdated);
```

Comment:
Should you remove the listener somewhere?

Reply:
One day, yes. We don't tear down the store at all though, and never change the field.